vpc.go

```diff
@@ -167,14 +167,17 @@ func (e *VPCExporter) collectVpcsPerRegionQuota(client *servicequotas.ServiceQuo
 func (e *VPCExporter) collectVpcsPerRegionUsage(ec2Svc *ec2.EC2, region string) {
 	ctx, cancelFunc := context.WithTimeout(context.Background(), e.timeout)
 	defer cancelFunc()
-	describeVpcsOutput, err := ec2Svc.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{})
+	numVpcs := 0
+	err := ec2Svc.DescribeVpcsPagesWithContext(ctx, &ec2.DescribeVpcsInput{}, func(page *ec2.DescribeVpcsOutput, lastPage bool) bool {
```

Comment:
Please create a function like this in the awsclient.go: https://github.com/app-sre/aws-resource-exporter/blob/master/pkg/awsclient/awsclient.go#L64

Reply:
Please also do that for all the other cases where you use Pages methods.

Reply:
I replaced all clients with the new facade and added all methods we still use to the interface (+ regenerated the mock).

Reply:
Thanks for doing that. Is there any chance you can add tests to the methods changed? :smile:

Reply:
I am not sure adding tests is that feasible for the `*Pages*` functions.
Those are using callbacks, which we would then have to mock out specially using some go-routines to push data into them, while waiting in the tests (which honestly, makes the test code much more complex than the code it tests ...).
So I added 2 basic tests for the non-callback functions, but I feel mocking those out will become too complex for the 5 lines of code it tests?